### PR TITLE
feat: host/port/login/pass as env var

### LINKED ISSUE #13: Configuration from Environment Variables
**Is your feature request related to a problem? Please describe.**
A request from the Slack server was to allow configuration of this plugin via Env Vars, similar to other plugins.

**Describe the solution you'd like**
Environment Variable based configuration to be added as an option.

**Describe alternatives you've considered**
Nope, seems pretty clear.

**Additional context**
[Slack Thread](https://app.slack.com/client/T01SW2PU554/C01UECB59A7/thread/C01UECB59A7-1657018731.252029)

### PR DESCRIPTION
**Pull request by graza-io**
Closes #13

### CODE REVIEW

File: imap/utils.go [hunk 1]
```diff
@@ -16,7 +18,18 @@ func login(ctx context.Context, d *plugin.QueryData) (*client.Client, error) {
 	port := 993
 	tlsEnabled := true
 	insecureSkipVerify := false
-	var host, login, password string
+
+	// Check env var settings
+	host := os.Getenv("IMAP_HOST")
```
Comment (cbruno10) [inline imap/utils.go:23]: @graza-io Can you please update `config/imap.spc` and `docs/index.md` mentioning these env variables? For `config/imap.spc`, can you please add a line like https://github.com/turbot/steampipe-plugin-aws/blob/ed9e5a8b2b463fb6e30bd2fc19bbb16ab9dc8607/config/aws.spc#L34? And for `docs/index.md`, can you please update the `Configuration` section like in https://hub.steampipe.io/plugins/turbot/aws#configuration?

File: imap/utils.go [hunk 2]
```diff
@@ -16,7 +18,18 @@ func login(ctx context.Context, d *plugin.QueryData) (*client.Client, error) {
...
+	login := os.Getenv("IMAP_LOGIN")
+	password := os.Getenv("IMAP_PASSWORD")
+
+	if portString, ok := os.LookupEnv("IMAP_PORT"); ok {
+		p, err := strconv.Atoi(portString)
+		if err == nil {
```
Comment (cbruno10) [inline imap/utils.go:29]: @graza-io I think we should check if they're passing a valid integer and a valid port number, like we have in https://github.com/turbot/steampipe-plugin-aws/blob/ed9e5a8b2b463fb6e30bd2fc19bbb16ab9dc8607/aws/service.go#L1163-L1165. Please let me know your thoughts.

**Review by cbruno10 — CHANGES_REQUESTED** [review]
@graza-io Thanks for opening this PR! If you have any questions, please let me know, thanks.

File: imap/utils.go (hunk 1 above)
Comment (graza-io) [reply imap/utils.go:23]: https://github.com/turbot/steampipe-plugin-imap/pull/18/commits/bf44bca7c39bf223105638347785be78046421ea

File: imap/utils.go (hunk 2 above)
Comment (graza-io) [reply imap/utils.go:29]: https://github.com/turbot/steampipe-plugin-imap/pull/18/commits/d3a33c5fc26bac51b7bb3f6fea5e1068b86978c2

**Review by cbruno10 — APPROVED** [review]

**Comment by cbruno10** [comment]
Thanks so much @graza-io , these are great additions!

**Pull request merged**